src/packet.cpp

```diff
@@ -0,0 +1,73 @@
+#include "packet.h"
+#include <openssl/md5.h>
+#include <iostream>
+#include <stdexcept>
+
+Packet::Packet(const std::array<uint8_t, 4096>& m_recvBuffer, size_t bytes)
+{
+    size_t length = m_recvBuffer[2] * 256 + m_recvBuffer[3];
```

Comment:
It would be nice to check that `size` is at least 20 bytes (the buffer contains the whole header).